wiki/Performance_Troubleshooting/en.md

```diff
@@ -0,0 +1,71 @@
+
+## The types of "lag"
+It is quite difficult to provide support when every kind of issue is grouped under the single word “lag”, so let’s start by defining some terms for specific scenarios.
+
+### Audio stuttering
+The actual audio track of the beatmap rewinds in time, or glitches out. Basically, when the game lags, you can hear it in the audio. When diagnosing or reporting, it may help to:
+
+- Check whether it happens on all available audio devices. If you're using a USB headset, try connecting via analog output on your PC instead.
+- Check whether turning on "Audio compatibility mode" in settings helps.
+
+### Spike frame
+During gameplay, everything on the screen stops for long enough to be detrimental to the gameplay experience. This may happen more than once, at random or at consistent intervals. When diagnosing or reporting, it may help to:
+
+- Count how many times it happens in an average play session.
+- Check whether it always happens.
+- Check how often it happens, and whether it's at a consistent interval.
+- Check whether it still happens straight after a PC restart.
+- Try closing any application which shows overlays on the game (Discord, Steam, etc.)
+
+### Input latency
+There is a noticeable delay between tapping on the keyboard or moving your mouse or tablet pen, and seeing a resulting action on the screen. When diagnosing or reporting, it may help to:
+
+- Check whether it happens on all input devices, or one specifically.
+- If you are using non-official drivers, try switching to official ones instead.
+
+### Loss of control
+During gameplay, one or more input devices stop responding. Everything on screen moves correctly but it’s not possible to play properly until control returns. When diagnosing or reporting, it may help to:
+
+- Press Ctrl-F11 to see the frame time display graph. Check whether there is a lot of white on the graph when this occurs (this means another app or driver on your system is using too much CPU time, delaying input).
+
+### Low frame rate
+During gameplay, the frame rate is unable to keep up, resulting in jittery and awkward movement of objects on the screen. This can also result in higher perceived input latency. When diagnosing or reporting, it may help to:
+
+- Check whether it still happens straight after a PC restart.
+- Try closing any application which shows overlays on the game (Discord, Steam, etc.).
+- Make sure you're not forcing vsync on at your display drivers (it should be set to "off" or "application preference").
+- Try changing frame limiter and see if it improves the situation.
+- Make sure you are running osu! in exclusive (not borderless) fullscreen mode. This will always improve performance.
+
+## Frame Limiters
+osu! offers multiple choices of frame limiters. Each of these have pros and cons. Note that regardless of setting, osu! will limit frame rate at menus to conserve on CPU usage.
```

Comment:
```suggestion
osu! offers multiple choices of frame limiters. Each of these has pros and cons. Note that regardless of setting, osu! will limit frame rate at menus to conserve on CPU usage.
```

Reply:
wow, english is hard. the first actually feels correct to me   yt logically i believe you are right.

Reply:
it's best as-is because "these" is plural. "each" by itself is *technically* singular but it can be skewed into plural depending on what type of noun comes after it. in this case you're comparing "these have" to "these has"

english is hard